File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
import sys, math

m, n = map(int, sys.stdin.readline().split())

def find_primes(maxVal):
    primes = [ True for _ in range(maxVal+1) ]
    primes[0] = primes[1] = False

    end = int(math.sqrt(maxVal))

    for i in range(2, end+1):
        if primes[i]:
            for j in range(i*i, maxVal+1, i):
                primes[j] = False

    return primes

primes = find_primes(n)

for i in range(m, n+1):
    if primes[i]:
        print(i)

File: 백준/Silver/1929. 소수 구하기/소수 구하기.js
const fs = require("fs");
let inputs = fs.readFileSync("/dev/stdin").toString().split(" ");
const m = parseInt(inputs[0]);
const n = parseInt(inputs[1]);

function setPrimes(maxNum) {
    let primeDatas = Array.from({length: maxNum+1}, () => true);
    primeDatas[0] = primeDatas[1] = false;
    for(let i = 2; i*i <= maxNum; i++) {
        if(primeDatas[i]) {
            for(let j = i*i; j <= maxNum; j += i) primeDatas[j] = false;
        }
    }
    return primeDatas;
}

const primes = setPrimes(n);
let primeNums = [];
for(let i = m; i <= n; i++) {
    if(primes[i]) primeNums.push(i);
}
console.log(primeNums.join('\n'));